internal/services/automation/automation_runbook_resource.go

```diff
@@ -330,31 +341,15 @@ func resourceAutomationRunbookCreateUpdate(d *pluginsdk.ResourceData, meta inter
 
 	d.SetId(id.ID())
 
-	automationAccountId := jobschedule.NewAutomationAccountID(subscriptionID, id.ResourceGroupName, id.AutomationAccountName)
-
-	jsIterator, err := jsClient.ListByAutomationAccountComplete(ctx, automationAccountId, jobschedule.ListByAutomationAccountOperationOptions{})
-	if err != nil {
-		return fmt.Errorf("loading Automation Account %q Job Schedule List: %+v", id.AutomationAccountName, err)
-	}
-
-	for _, item := range jsIterator.Items {
-		if itemProps := item.Properties; itemProps != nil {
-			if itemProps.Runbook != nil && itemProps.Runbook.Name != nil && *itemProps.Runbook.Name == id.RunbookName {
-				if itemProps.JobScheduleId == nil || *itemProps.JobScheduleId == "" {
-					return fmt.Errorf("job schedule Id is nil or empty listed by %s Job Schedule List: %+v", id, err)
-				}
-				parsedId := jobschedule.NewJobScheduleID(subscriptionID, id.ResourceGroupName, id.AutomationAccountName, *itemProps.JobScheduleId)
-				if resp, err := jsClient.Delete(ctx, parsedId); err != nil {
-					if !response.WasNotFound(resp.HttpResponse) {
-						return fmt.Errorf("deleting job schedule Id listed by %s Job Schedule List:%v", id, err)
-					}
-				}
-			}
+	// **don't need** to list job schedules and delete all of them. update the runbook will recreate these job schedules automatically,
+	// but with a different job schedule id
+	// Fix issue: https://github.com/hashicorp/terraform-provider-azurerm/issues/8634
+	if jsValue, ok := d.GetOk("job_schedule"); (ok && d.IsNewResource()) || d.HasChange("job_schedule") {
+		if err = deleteLinkedJobSchedules(ctx, jsClient, &id); err != nil {
+			return fmt.Errorf("clean existing job schedule links: %v", err)
 		}
-	}
 
-	if v, ok := d.GetOk("job_schedule"); ok {
```

Comment:
The `job_schedule` is not documented in the `azurerm_automation_job_schedule` (apprently was missing in #7555). This is a good chance to add it up. Be note that we shall mention that users should either use the `job_schedule` property in this resource, or use the `azurerm_automation_job_schedule`, but not both. See [this](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/subnet#:~:text=NOTE%20on%20Virtual%20Networks%20and%20Subnet%27s%3A) as a reference.